pkg/tfbridge/schema.go

```diff
@@ -661,6 +678,9 @@ func (ctx *conversionContext) applyDefaults(
 		if _, exactlyOneOfConflicts := exactlyOneOf[name]; exactlyOneOfConflicts {
 			continue
 		}
+		if _, requiredWithConflict := requiredWith[name]; requiredWithConflict {
+			continue
+		}
```

Comment:
I'm concerned that not applying defaults to fields that have `requiredWith` set will cause a regression for existing providers.

Given that we are aware of the requiredWith properties, can we apply defaults if and only if it won't error.

Something like
```suggestion
		for p := range requiredWith[name] {
			if _, hasP := result[p]; !hasP {
				// We are missing a property `p` that would become required if `name` was set. To give the deployment the best change of succeeding, don't apply the default here.
				continue
			}
		}
```

Reply:
Great proposal, thanks! Do you think we should think about defaults for the linked property here too? E.g. this property might have a default and the `RequireWith` property could have a default as well, so if we set both defaults it might all work out.

A slight complication here is that if a property with a default has a validation function and the default does not pass the validation we (or tf, not sure) actually don't set that default.